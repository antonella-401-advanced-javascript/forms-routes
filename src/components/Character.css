.Character {
  width: 80%;
  max-width: 300px;
  margin: auto;

  div {
    margin: auto;
    width: 90%;
    max-width: 400px;
    max-height: 390px;
  }
  img {
    width: 100%;
    border-radius: 3px;
    margin: 0;
  }
  p {
    text-align: center;
    color: whitesmoke;
    font-weight: bold;
  }
}