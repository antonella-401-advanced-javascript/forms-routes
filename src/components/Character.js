import React from 'react';
import PropTypes from 'prop-types';
import styles from './Character.css';

const Character  = ({ character }) => {
  return (
    <div className={styles.Character}>
      <div>
        <img src={character.photoUrl}/>
      </div>
      <p>{character.name}</p>
    </div>
  );
};

Character.propTypes = {
  character: PropTypes.object
};

export default Character;