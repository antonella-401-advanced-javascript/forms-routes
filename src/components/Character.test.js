import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders Character', () => {
    const testCharacter = {
      photoUrl: 'https://www.outbrain.com/techblog/wp-content/uploads/2017/05/road-sign-361513_960_720.jpg',
      name: 'test'
    };
    const wrapper = shallow(<Character character={testCharacter} />);
    expect(wrapper).toMatchSnapshot();
  });
});