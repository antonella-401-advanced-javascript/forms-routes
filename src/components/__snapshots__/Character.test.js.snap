// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<div
  className="Character"
>
  <div>
    <img
      src="https://www.outbrain.com/techblog/wp-content/uploads/2017/05/road-sign-361513_960_720.jpg"
    />
  </div>
  <p>
    test
  </p>
</div>
`;
